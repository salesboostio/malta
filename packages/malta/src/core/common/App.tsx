import * as React from "react";
import styled, { ThemeProvider } from "styled-components";

const Wrapper = styled.div`
  min-height: 100%;
  width: 100%;
`;

interface Props {
  theme: object;
  children?: any;
}

export const App = (props: Props) => {
  return (
    <ThemeProvider theme={props.theme}>
      <Wrapper>
        {props.children}
      </Wrapper>
    </ThemeProvider>
  );
};
