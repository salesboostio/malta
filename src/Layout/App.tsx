import * as React from "react";
import * as classNames from "classnames";
import styled from "styled-components";
import {
  $primaryBackground
} from "../_variables/colors";

export interface IProps {
  prefixCls?: string;
  className?: string;
}

const Layout = styled.div`
  background: ${$primaryBackground};
  padding: 0;
  margin: 0;
  display: block;
`;

export class App extends React.Component<IProps, any> {
  constructor(props: IProps) {
    super(props);
  }

  public render() {
    const {
      prefixCls = "sb-app",
      className,
      ...props
    } = this.props;

    const classString = classNames(prefixCls, className);

    return (
      <Layout className={classString} {...props}></Layout>
    );
  }
}
