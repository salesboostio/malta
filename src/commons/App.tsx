import * as React from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "./theme";

const Wrapper = styled.div`
  height: 100%;
`;

export class App extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
  }

  public render() {
    return (
      <ThemeProvider theme={theme}>
        <Wrapper>
          {this.props.children}
        </Wrapper>
      </ThemeProvider>
    );
  }
}
