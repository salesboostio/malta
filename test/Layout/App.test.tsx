
import * as React from "react";
import { create } from "react-test-renderer";
import { App } from "../../src/Layout/App";

test("App", () => {
    const node = create(<App />).toJSON();
    expect(node).toMatchSnapshot();
});
