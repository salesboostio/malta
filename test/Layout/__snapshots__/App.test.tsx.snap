// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<div
  className="sb-app sc-bdVaJa dVZBp"
/>
`;
